__init__.py

```diff
@@ -12,314 +12,420 @@
+        self.enable_intent("handle_status_timer")
+
+        # reset the mute flag with a new timer
+        self.mute = False
+
+    def _get_next_timer(self):
+        # Retrieve the next timer set to trigger
+        next = None
+        for timer in self.active_timers:
+            if not next or timer["expires"] < next["expires"]:
+                next = timer
+        return next
+
+    def _get_timer(self, name):
+        # Retrieve the next timer set to trigger
+        for timer in self.active_timers:
+            if timer["name"] == name:
+                return timer
+        return None
+
+    def update_display(self, message):
+        # Get the next triggering timer
+        timer = self._get_next_timer()
+        if not timer:
+            # No active timers, clean up
+            self.cancel_scheduled_event('ShowTimer')
+            self.displaying_timer = None
+            self.disable_intent("handle_cancel_timer")
+            self.disable_intent("handle_mute_timer")
+            self.disable_intent("handle_status_timer")  # TODO: Change to Adapt for this
+            self.enclosure.eyes_reset()
+            self._stop_beep()
+            self._clear_display()
+            return
 
-    def handle_status_timer(self, message):
-        """ callback for timer status intent
+        # Check if the display frequency is set correctly for the
+        # closest timer...
+        if timer != self.displaying_timer:
+            self.cancel_scheduled_event('ShowTimer')
+            # Create a callback with appropriate frequency
+            freq = timer["duration"]/100.0
+            if freq < 1:
+                freq = 1
+            self.schedule_repeating_event(self.update_display,
+                                          None, freq,
+                                          name='ShowTimer')
+            self.displaying_timer = timer
+
+        # Calc remaining time and show using faceplate
+        now = datetime.now()
+
+        if (timer["expires"] > now):
+            # Timer still running
+            remaining = (timer["expires"] - now).seconds
+            dur = timer["duration"]
+            self.eyes_fill(int(remaining*100.0/dur))
+        else:
+            # Timer has expired but not been cleared, flash eyes
+            overtime = (now - timer["expires"]).seconds
+            if overtime % 2 == 0:
+                self.enclosure.eyes_off()
+            else:
+                self.enclosure.eyes_on()
 
-            Args:
-                message (Message): object passed by messagebus
-        """
+            # beep every 10 seconds
+            if overtime % 10 == 0 and not self.mute:
+                self._play_beep()
+
+            # Show the expired time.  This naturally "flashes"
+            self._show(nice_duration(overtime, speech=False))
+
+    # Handles 'How much time left'
+    @intent_file_handler('status.timer.intent')
+    def handle_status_timer(self, message):
```

Comment:
when setting multiple timers, mycroft will only give you status of the most recently set timer, and not the others even when you explicitly ask for it. It seems to hit the handle_stat_timer intent though nothing is spoken.

Reply:
I've reset the skills service just in case something was wonky on load. It seems that asking for "status of the timer", then explicitly say "5 minute timer" afterward works, but saying "status of the 5 minute timer" does not if you set another timer after the 5 minute timer.

Reply:
Release 0.9.17 broke several things.  There is a pending mycroft-core fix for this, but I think I have it working for now.  Will need to remove the "hack" after next release.